### PR TITLE
[AVR-763] Perform flip in the render_context

### PR DESCRIPTION
**Pull request by lucabergamini**
Perform flip of y axis in the `render_contex` (while creating the matrix)
Things to do:
- [x] add config flag to preserve previous models
- [x] read the config value and pass it to the render_context
- [x] fix tests ~(should be just the one with the render_context itself)~ spoiler: it was not
- [x] fix notebooks
- [x] train new models
- [x] test new models
- [x] upload new models

I've removed a comment in the satellite, maybe I should replace it with something else

### CODE REVIEW

File: examples/agent_motion_prediction/agent_motion_config.yaml [hunk 1]
```diff
@@ -43,6 +43,11 @@ raster_params:
   # whether to completely disable traffic light faces in the semantic rasterizer
   disable_traffic_light_faces: False
 
+  # when set to True performs a flip inside the rasteriser.
```
Comment (yawei-ye) [inline examples/agent_motion_prediction/agent_motion_config.yaml:46]:
```suggestion
  # When set to True, the rasterizer will set the raster origin at bottom left, i.e. vehicles are driving on the right side of the road. With this change, the vertical flipping on the raster used in the visualization code is no longer needed.
  # Note that this will flip the model input raster and break previously trained models. 
  # If users want to restore the old behavior, please set the "set_origin_to_bottom" to False. Additional, users will also need to restore the vertical flipping on the raster in the visualisation code.
```

File: examples/planning/config.yaml [hunk 2]
```diff
@@ -43,6 +43,11 @@ raster_params:
   # whether to completely disable traffic light faces in the semantic rasterizer
   disable_traffic_light_faces: False
 
+  # when set to True performs a flip inside the rasteriser.
```
Comment (yawei-ye) [inline examples/planning/config.yaml:46]: Same change as in `agent_motion.config`

File: l5kit/l5kit/rasterization/rasterizer_builder.py [hunk 3]
```diff
@@ -126,6 +126,7 @@ def build_rasterizer(cfg: dict, data_manager: DataManager) -> Rasterizer:
         raster_size_px=np.array(raster_cfg["raster_size"]),
         pixel_size_m=np.array(raster_cfg["pixel_size"]),
         center_in_raster_ratio=np.array(raster_cfg["ego_center"]),
+        origin_bottom=raster_cfg["origin_bottom"],
```
Comment (yawei-ye) [inline l5kit/l5kit/rasterization/rasterizer_builder.py:129]:
```suggestion
        vertical_flip=raster_cfg["set_origin_to_bottom"],
```

Comment (yawei-ye) [reply l5kit/l5kit/rasterization/rasterizer_builder.py:129]: `origin_bottom` is unclear, what about `vertical_flip`?

File: l5kit/l5kit/rasterization/render_context.py [hunk 4]
```diff
@@ -24,12 +30,15 @@ def __init__(
...
         self.center_in_raster_ratio = center_in_raster_ratio
+        self.origin_bottom = origin_bottom
 
         scaling = 1.0 / pixel_size_m  # scaling factor from world to raster space [pixels per meter]
         center_in_raster_px = center_in_raster_ratio * raster_size_px
         self.raster_from_local = np.array(
             [[scaling[0], 0, center_in_raster_px[0]], [0, scaling[1], center_in_raster_px[1]], [0, 0, 1]]
         )
+        if origin_bottom:
+            self.raster_from_local = flip_y_axis(self.raster_from_local, self.raster_size_px[1])
```
Comment (yawei-ye) [inline l5kit/l5kit/rasterization/render_context.py:41]:
```suggestion
            self.raster_from_local = vertical_flip(self.raster_from_local, self.raster_size_px[1])
```
I think `vertical_flip` is more clear than `flip_y_axis`: https://www.mathsisfun.com/definitions/vertical-flip.html#:~:text=more%20...,vertical%20direction%20(up%2Ddown)

File: l5kit/l5kit/rasterization/rasterizer_builder.py (hunk 3 above)
Comment (lucabergamini) [reply l5kit/l5kit/rasterization/rasterizer_builder.py:129]: I don't think it's more clear, because both can be seen as the flip of the other, but I'm fine in changing it

File: l5kit/l5kit/rasterization/render_context.py (hunk 4 above)
Comment (lucabergamini) [reply l5kit/l5kit/rasterization/render_context.py:41]: probably, but given I change the name of the variable to `vertical_flip` I can't change this one to the same name

File: l5kit/l5kit/rasterization/rasterizer_builder.py (hunk 3 above)
Comment (yawei-ye) [reply l5kit/l5kit/rasterization/rasterizer_builder.py]: Agree, how about rename  parameter `origin_bottom` to `set_origin_to_bottom`?

File: l5kit/l5kit/rasterization/render_context.py (hunk 4 above)
Comment (yawei-ye) [reply l5kit/l5kit/rasterization/render_context.py:41]: How about name the variable as `set_origin_to_bottom ` and rename the function name to `vertical_flip` or `vflip` same as torchvision.transform: https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.functional.vflip

File: examples/visualisation/visualise_data.ipynb [hunk 5]
```diff
@@ -296,13 +296,10 @@
    "source": [
     "## System Origin and Orientation\n",
     "\n",
-    "At this point you may have noticed that we flip the image on the **Y-axis** before plotting it.\n",
+    "~At this point you may have noticed that we flip the image on the **Y-axis** before plotting it.~\n",
```
Comment (yawei-ye) [inline examples/visualisation/visualise_data.ipynb:299]:
```suggestion
    "~At this point you may have noticed that we vertically flip the image before plotting it.~\n",
```

File: examples/visualisation/visualise_data.ipynb [hunk 6]
```diff
@@ -296,13 +296,10 @@
...
     "\n",
-    "When moving from 3D to 2D we stick to a right-hand system, where the origin is in the bottom-left corner with positive x-values going right and positive y-values going up the image plane. The camera is facing down the negative z axis. \n",
+    "Flip of the Y-axis is not required anymore as it's already performed inside the rasteriser.\n",
```
Comment (yawei-ye) [inline examples/visualisation/visualise_data.ipynb:301]:
```suggestion
    "Vertical flipping is not required anymore as it's already performed inside the rasteriser.\n",
```

**Review by yawei-ye — DISMISSED** [review]
Minor, otherwise LGTM

File: l5kit/l5kit/tests/rasterization/render_context_test.py [hunk 7]
```diff
@@ -14,11 +16,14 @@ def test_transform_points_to_raster() -> None:
         raster_size_px=image_shape_px,
         pixel_size_m=pixel_size_m,
         center_in_raster_ratio=center_in_raster_ratio,
-        set_origin_to_bottom=False,
+        set_origin_to_bottom=set_origin_to_bottom,
     )
 
     input_points = np.array([[0, 0], [10, 10], [-10, -10]])
-    expected_output_points = np.array([[100, 102], [110, 112], [90, 92]])
+    if not set_origin_to_bottom:
```
Comment (yawei-ye) [inline l5kit/l5kit/tests/rasterization/render_context_test.py:23]: Minor
```suggestion
    if set_origin_to_bottom:
```
And swap the Line 24 & 26

**Review by yawei-ye — COMMENTED** [review]
Minor, otherwise LGTM

**Review by yawei-ye — APPROVED** [review]
LGTM! thanks!
